packages/common/src/ContractInteractor.ts

```diff
@@ -940,10 +985,73 @@ calculateTransactionMaxPossibleGas: result: ${result}
     return await this.relayHubInstance.penalizer()
   }
 
+  private async _hubRelayRegistrarAddress (): Promise<Address> {
+    return await this.relayHubInstance.relayRegistrar()
+  }
+
   penalizerAddress (): Address {
     return this.penalizerInstance.address
   }
 
+  /**
+   * discover registered relays
+   * @param subset if set, then filter only to these relays
+   */
+  async getRegisteredRelays (subset?: string[], fromBlock?: number): Promise<RelayRegisteredEventInfo[]> {
```

Comment:
1. the code is shared by the client and relayer. I could introduce a new registrar wrapper to be used by both.
2. the logic is not new, but its simpler: instead of first search for every event of relayer, and then a 2nd run just for the register, I do a single run to check just for the register.

Reply:
I see. I would still keep it outside of ContractInteractor, that class is growing like crazy :-)